machineconfiguration/v1alpha1/types_machineconfigstate.go

```diff
@@ -0,0 +1,125 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineConfigState describes the health of the Machines on the system
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type MachineConfigState struct {
```

Comment:
If this is going to map 1:1 with a Node, how would you feel about calling it a `MachineConfigNode`, since that would then align with others (eg `CSINode`) who have the same pattern

Reply:
Yeah, I like that name and it fits with names like `MachineConfigPool`